### ISSUE TITLE
reportGeneralTypeIssues when using `@property` and `@abstractmethod`

### ISSUE THREAD

**Issue by yamap55** [microsoft/pyright#2253]
**Describe the bug**

raise error when using `@property` and `@abstractmethod`.

```
error: Expression of type "Literal['cola']" cannot be assigned to declared type "property"
    "Literal['cola']" is incompatible with "property" (reportGeneralTypeIssues)
```

**To Reproduce**

```
pyright /project/drink.py
```

**Expected behavior**
not raise.

**Screenshots or Code**

```python
from abc import ABC, abstractmethod
class Drink(ABC):
    @property
    @abstractmethod
    def name(self) -> str:
        pass

class Cola(Drink):
    name = "cola"
```

**VS Code extension or command-line**

```
$ pyright --version
pyright 1.1.165
```

**Additional context**
Setting a type hint on an implemented property does not raise an exception.
If this was the correct behavior, I would like to ask why.

```python
from abc import ABC, abstractmethod
class Drink(ABC):
    @property
    @abstractmethod
    def name(self) -> str:
        pass

class Cola(Drink):
    name:str = "cola" # add type hint
```

- GitHub Actions log
  - https://github.com/AtsushiOzawa/tdd_vending-machine/runs/3482653202

**Comment by yamap55** [comment]
sorry. I didn't know "typing.Literal".
https://docs.python.org/3/library/typing.html#typing.Literal

```python
from abc import ABC, abstractmethod
from typing import Literal
class Drink(ABC):
    @property
    @abstractmethod
    def name(self) -> Literal:
        pass

class Cola(Drink):
    name:str = "cola"
```

**Comment by yamap55** [comment]
issue reopen because the error was not resolved. 🙇

**Comment by erictraut** [comment]
You are attempting to override a property with an attribute. This is sometimes done in Python code, but it's really not type safe because the semantics of a property are somewhat different from an attribute.

The preferred approach here is to override a property with a property, like this:

```python
class Cola(Drink):
    @property
    def name(self) -> str:
        return "cola"
```

If you want to override the property with an attribute, you need to declare a new type for the symbol in the child class.

```python
    name: str = "cola"
```

**Comment by yamap55** [comment]
Thank you electrart.
I understand this specification.
